localstack/services/cloudformation/models/iam.py

```diff
@@ -221,16 +348,110 @@ def get_physical_resource_id(self, attribute=None, **kwargs):
 
     @staticmethod
     def get_deploy_templates():
+        def _add_roles(resource_id, resources, resource_type, func, stack_name):
+            client = aws_stack.connect_to_service("iam")
+            resource = resources[resource_id]
+            props = resource["Properties"]
+            if props.get("Roles", []):
+                client.add_role_to_instance_profile(
+                    InstanceProfileName=props["InstanceProfileName"],
+                    RoleName=props["Roles"][0],
+                )
+
         return {
-            "create": {
-                "function": "create_instance_profile",
-                "parameters": {
-                    "InstanceProfileName": "InstanceProfileName",
-                    "Path": "Path",
+            "create": [
+                {
+                    "function": "create_instance_profile",
+                    "parameters": {
+                        "InstanceProfileName": "InstanceProfileName",
+                        "Path": "Path",
+                    },
                 },
-            },
+                {"function": _add_roles},
+            ],
             "delete": {
                 "function": "delete_instance_profile",
                 "parameters": {"InstanceProfileName": "InstanceProfileName"},
             },
         }
+
+
+class IAMGroup(GenericBaseModel):
+    @staticmethod
+    def cloudformation_type():
+        return "AWS::IAM::Group"
+
+    def get_physical_resource_id(self, attribute=None, **kwargs):
+        return self.props.get("GroupName")
+
+    def get_resource_name(self):
```

Comment:
Note to self: We need to evaluate whether we still need `get_resource_name(..)` at all - I think we may be able to get rid of it.

Looking at the template_deployer.py code, it seems to be only used for:
1. to call moto's `create_from_cloudformation_json` (which we'll strip out soon anyway), and
2. to replace instances of `PLACEHOLDER_RESOURCE_NAME` (where we discussed that it could be replaced with some sort of resource-specific `add_default_props(..)`).

(No change required for this PR, but to be discussed..)